lib/saml/user.rb

```diff
@@ -10,113 +10,89 @@ module SAML
-                    LOA::MAPPING.invert[3] => 'idproof',
-                    'multifactor' => 'multifactor',
-                    'dslogon_multifactor' => 'dslogon_multifactor',
-                    'myhealthevet_multifactor' => 'myhealthevet_multifactor' }.freeze
-    UNKNOWN_CONTEXT = 'unknown'
+    AUTHN_CONTEXTS = {
+      LOA::IDME_LOA1 => { loa_current: '1', sign_in: { service_name: 'idme' } },
+      LOA::IDME_LOA3 => { loa_current: '3', sign_in: { service_name: 'idme' } },
+      'multifactor' => { loa_current: nil, sign_in: { service_name: 'idme' } },
+      'myhealthevet_multifactor' => { loa_current: nil, sign_in: { service_name: 'myhealthevet' } },
+      'myhealthevet_loa3' => { loa_current: '3', sign_in: { service_name: 'myhealthevet' } },
+      'dslogon_multifactor' => { loa_current: nil, sign_in: { service_name: 'dslogon' } },
+      'dslogon_loa3' => { loa_current: '3', sign_in: { service_name: 'dslogon' } },
+      'myhealthevet' => { loa_current: nil, sign_in: { service_name: 'myhealthevet' } },
+      'dslogon' => { loa_current: nil, sign_in: { service_name: 'dslogon' } }
+    }.freeze
 
     attr_reader :saml_response, :saml_attributes, :user_attributes
 
     def initialize(saml_response)
       @saml_response = saml_response
       @saml_attributes = saml_response.attributes
-      @user_attributes = user_attributes_class.new(saml_attributes, real_authn_context)
-      log_warnings_to_sentry!
+
+      Raven.extra_context(
+        saml_attributes: saml_attributes&.to_h,
+        saml_response: Base64.encode64(saml_response&.response || '')
+      )
+
+      @user_attributes = user_attributes_class.new(saml_attributes, authn_context)
+      log_warnings_to_sentry
     end
 
     def changing_multifactor?
-      return false if real_authn_context.nil?
-      real_authn_context.include?('multifactor')
+      return false if authn_context.nil?
+      authn_context.include?('multifactor')
     end
 
     def to_hash
       user_attributes.to_hash.merge(Hash[serializable_attributes.map { |k| [k, send(k)] }])
     end
 
-    def self.context_key(authn_context)
-      CONTEXT_MAP[authn_context] || UNKNOWN_CONTEXT
-    rescue StandardError
-      UNKNOWN_CONTEXT
-    end
-
     private
 
-    # we serialize user.rb with this value, in the case of everything other than mhv/dslogon,
-    # this will only ever be one of 'dslogon, mhv, or nil'
-    def authn_context
-      return 'dslogon' if dslogon?
-      return 'myhealthevet' if mhv?
-      nil
-    end
-
     # returns the attributes that are defined below, could be from one of 3 distinct policies, each having different
     # saml responses, hence this weird decorating mechanism, needs improved abstraction to be less weird.
     def serializable_attributes
       %i[authn_context]
     end
 
-    def dslogon?
-      saml_attributes.to_h.keys.include?('dslogon_uuid')
-    end
+    # NOTE: The actual exception, if any, should get raised when to_hash is called. Hence "suppress"
```

Comment:
it looks like you took out `suppress(Exception)` but mention it in the comment.

Reply:
fixed this